app/controllers/events_controller.rb

```diff
@@ -0,0 +1,59 @@
+class EventsController < ApplicationController
+    before_action :logged_in?, only: [ :new, :create, :edit, :update, :destroy]
```

Comment:
- [ ] Good job handling the unsinged user actions, but this actually does nothing. Our `logged_in` method just return true of false, but it should take an action if the user is unsigned in, like redirecting to login page.

So we can add a `require_login` method which checks if the user is `logged_in?` and redirect to sign in page if not `logged_in?`. Then use it instead of `logged_in?` `before_action`